internal/job/backup/backup_test.go

```diff
@@ -491,6 +491,60 @@ func Test_IncrementalBackup_Success_Resume(t *testing.T) {
 	})
 }
 
+func Test_FullBackup_Success_DifferentNode(t *testing.T) {
+	// CSATEST-1496
+	// Description:
+	//   Create a full backup with no error on a node different from its previous backup.
+	//
+	// Arrange:
+	//   - Set SourceCandidateSnapshotID to an non-empty value.
+	//   - Set backup_metadata table to be empty.
+	//
+	// Act:
+	//   Run the backup process to create an full backup.
+	//
+	// Assert:
+	//   Check if the contents of the full backup is correct.
+
+	// Arrange
+	cfg := setup(t, &setupInput{})
+	require.NotNil(t, cfg.incrementalBackupInput.SourceCandidateSnapshotID)
+
+	// Act
+	backup := NewBackup(cfg.incrementalBackupInput /* Use incrementalBackupInput to set SourceCandidateSnapshotID */)
+	err := backup.Perform()
+	require.NoError(t, err)
+
+	// Assert
+	testutil.AssertActionPrivateDataIsEmpty(t, cfg.finRepo, cfg.incrementalBackupInput.ActionUID)
+	rawImage, err := fake.ReadRawImage(cfg.nlvRepo.GetRawImagePath())
+	assert.NoError(t, err)
+
+	for _, diff := range rawImage.AppliedDiffs {
+		assert.Equal(t, cfg.incrementalBackupInput.TargetRBDPoolName, diff.PoolName)
+		assert.Nil(t, diff.FromSnap)
```

Comment:
Since this test case is backing up the volume with the specified snapshot ID, I think the length of AppliedDiff is 2 and one of them has FromSnap set. However, I am not confident about the behavior of fake, so please let me know if I am wrong.